src/engine/contactsengine.cpp

```diff
@@ -262,6 +262,7 @@ class ContactFetchJob : public TemplateJob<QContactFetchRequest>
         QList<QContact> contacts;      {
             QMutexLocker locker(mutex);
             contacts = m_contacts;
+            m_contacts.clear();
```

Comment:
Can updateState() happen after update()?  If so, I think we cannot clear m_contacts here.  Certainly, I don't think we should need to.  The more I think about this, the more I think it must be a bug in QDeclarativeContactModel.